File: Basic/26.Write a program to find the largest number possible from an array of numbers/Solution.py
def largest_number(nums):
    from functools import cmp_to_key

    # Custom comparator
    def compare(x, y):
        if x + y > y + x:
            return -1
        elif x + y < y + x:
            return 1
        else:
            return 0

    nums = sorted(nums, key=cmp_to_key(compare))
    return ''.join(nums)

# Input from user
n = int(input("Enter the number of elements: "))
nums = [input("Enter number: ") for _ in range(n)]

# Find and print the largest number
print("Largest number:", largest_number(nums))

File: Basic/26.Write a program to find the largest number possible from an array of numbers/Solution.java
import java.util.*;

public class LargestNumber {
    // Custom comparator
    static class CustomComparator implements Comparator<String> {
        @Override
        public int compare(String x, String y) {
            return (y + x).compareTo(x + y); // Sort in descending order
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();

        String[] nums = new String[n];
        System.out.println("Enter the numbers:");
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.next();
        }

        Arrays.sort(nums, new CustomComparator());

        System.out.print("Largest number: ");
        for (String num : nums) {
            System.out.print(num);
        }
        System.out.println();
    }
}
